gc/verbose/handler_standard/VerboseHandlerOutputStandard.cpp

```diff
@@ -801,29 +801,23 @@ MM_VerboseHandlerOutputStandard::hasOutputMemoryInfoInnerStanza()
 }
 
 void
-MM_VerboseHandlerOutputStandard::outputMemType(MM_EnvironmentBase* env, uintptr_t indent, const char* type, uintptr_t free, uintptr_t total, uintptr_t microFragment, uintptr_t macroFragment)
+MM_VerboseHandlerOutputStandard::outputMemType(MM_EnvironmentBase* env, uintptr_t indent, const char* type, uintptr_t free, uintptr_t total, uint32_t tenureFragmentation, uintptr_t microFragment, uintptr_t macroFragment)
 {
-	MM_VerboseWriterChain* writer = _manager->getWriterChain();
-
-	writer->formatAndOutput(
-			env,
-			indent,
-			"<mem type=\"%s\" free=\"%zu\" total=\"%zu\" percent=\"%zu\" micro-fragmented=\"%zu\" macro-fragmented=\"%zu\" />",
-			type, free, total,
-			((total == 0) ? 0 : ((uintptr_t)(((uint64_t)free*100) / (uint64_t)total))), microFragment, macroFragment);
-}
+	char memInfoBuffer[INITIAL_BUFFER_SIZE] = "";
+	char* bufferPtr = memInfoBuffer;
 
-void
-MM_VerboseHandlerOutputStandard::outputMemType(MM_EnvironmentBase* env, uintptr_t indent, const char* type, uintptr_t free, uintptr_t total)
-{
 	MM_VerboseWriterChain* writer = _manager->getWriterChain();
 
-	writer->formatAndOutput(
-			env,
-			indent,
-			"<mem type=\"%s\" free=\"%zu\" total=\"%zu\" percent=\"%zu\" />",
-			type, free, total,
-			((total == 0) ? 0 : ((uintptr_t)(((uint64_t)free*100) / (uint64_t)total))));
+	bufferPtr += sprintf(bufferPtr, "<mem type=\"%s\" free=\"%zu\" total=\"%zu\" percent=\"%zu\"",
```

Comment:
You should be using omrstr_printf() and not sprintf().  Can you make the appropriate changes to use omrstr_printf().  Also are you checking to make sure the buffer is not overflowed?